src/screens/ProfileScreen/index.js

```diff
@@ -629,23 +629,25 @@ const ProfileScreen = ({route}) => {
   };
 
   const onHeaderOptionClicked = (post) => {
+    console.log('post', post);
     setSelectedPostForOption(post);
     setIsOptionModalOpen(true);
   };
 
-  const removePostByIdFromContext = () => {
-    const deletedIndex = feeds?.findIndex((find) => selectedPostForOption?.id === find?.id);
+  const removePostByIdFromContext = (post) => {
+    const deletedIndex = feeds?.findIndex((find) => post?.id === find?.id);
     const newData = [...feeds];
     newData?.splice(deletedIndex, 1);
     setMyProfileFeed(newData, myProfileDispatch);
   };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `removePostByIdFromContext` function doesn't handle the case when `deletedIndex` is -1, which means the post was not found in the feeds. This could lead to unexpected behavior as splice with -1 will remove the last element from the array. You should add a check for this case.
```diff
+ if (deletedIndex !== -1) {
  const newData = [...feeds];
  newData?.splice(deletedIndex, 1);
  setMyProfileFeed(newData, myProfileDispatch);
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->